azurelinuxagent/ga/exthandlers.py

```diff
@@ -423,6 +423,14 @@ def handle_ext_handler(self, ext_handler, etag):
 
         try:
             state = ext_handler.properties.state
+
+            self.get_artifact_error_state.reset()
```

Comment:
the error state should be reset after the "if decide_version" (because download succeeded)

now, since with this change we short-circuit the logic if there is not a new goal state, it seems to me that the error state is not needed and we should always report errors (since it is a new goal state) -- could you review the code to check if this is true?

Reply:
good point, I didnt really check this part. Let me take a look and verify the need for it

Reply:
Done, removed the error state completely